### PR TITLE
feat(memory): LifecycleManager and CleanupManager (task-384, task-385, task-393)

### PR DESCRIPTION
**Pull request by jpoley**
## Summary

Lifecycle management for Task Memory - automatic creation/archival on task state changes.

## Tasks Completed
- **task-384**: LifecycleManager component ✅ (6/7 ACs - diagram is separate doc task)
- **task-385**: CLI hooks integration (2/7 ACs - hooks ready, CLI wiring pending upstream)
- **task-393**: CleanupManager component ✅ (7/7 ACs)

## Files Added
- `src/specify_cli/memory/lifecycle.py` - State transition handling
- `src/specify_cli/memory/cleanup.py` - Archive/delete operations
- `src/specify_cli/memory/hooks.py` - CLI integration hooks
- `tests/test_memory_lifecycle.py` - 35 tests
- `tests/test_memory_cleanup.py` - 27 tests

### task-384: LifecycleManager
- [x] Implement LifecycleManager class
- [x] Hook into task state transitions (To Do→In Progress→Done→Archive)
- [x] Handle rollback scenario (Done→In Progress)
- [x] Update backlog/CLAUDE.md with active task @import
- [x] Comprehensive unit tests (35 tests, 100% coverage)
- [x] Error handling tested
- [ ] Document lifecycle state machine (separate task)

### task-385: CLI Integration
- [ ] Modify backlog CLI (requires upstream changes)
- [x] on_task_status_change() hook implemented
- [x] old/new state parameters supported
- [ ] CLI output messages (pending upstream)
- [ ] Live CLI testing (pending upstream)
- [ ] Integration tests (pending upstream)

### task-393: CleanupManager
- [x] CleanupManager class implemented
- [x] archive_older_than(days) method
- [x] delete_archived_older_than(days) method
- [x] Dry-run mode support
- [x] Logging for all operations
- [x] Tested with various thresholds
- [x] Unit tests (27 tests, 90% coverage)

## Test Plan
- [x] All 62 memory tests pass
- [x] 81% overall coverage, 100% on LifecycleManager
- [x] State transitions work correctly
- [x] CLAUDE.md auto-updates verified
- [x] Linting: All checks passed
- [x] Edge cases: unicode, concurrent ops, error handling

## Validation Results
- **QA Guardian**: APPROVED - Exceptional quality, 62 tests, 100% critical path coverage
- **Security Review**: PASS with note - Path traversal protection recommended for defense-in-depth (task IDs are internally generated, low risk)

## Known Limitations
- task-385 requires upstream backlog CLI integration to complete ACs 1, 4-7
- State machine diagram documentation deferred to task-380

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/specify_cli/memory/cleanup.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+        active_count = len(active_tasks)
+
+        oldest_active_days = 0
+        if active_tasks:
+            oldest_active_mtime = min(
+                datetime.fromtimestamp(self.store.get_path(task_id).stat().st_mtime)
+                for task_id in active_tasks
+                if self.store.get_path(task_id).exists()
+            )
+            oldest_active_days = (now - oldest_active_mtime).days
```
[the remark this answers is not in the document]
Comment (jpoley) [reply src/specify_cli/memory/cleanup.py]: fix me

File: tests/test_memory_cleanup.py [hunk 2]
```diff
@@ -0,0 +1,511 @@
+"""Tests for CleanupManager component."""
+
+import pytest
+import time
```
[the remark this answers is not in the document]
Comment (jpoley) [reply tests/test_memory_cleanup.py]: fix me
